pkg/cache/cluster.go

```diff
@@ -666,8 +656,10 @@ func (c *clusterCache) watchEvents(ctx context.Context, api kube.APIResourceInfo
 
 				obj, ok := event.Object.(*unstructured.Unstructured)
 				if !ok {
+					resourceVersion = ""
```

Comment:
I'd like to see if we can get away without resetting this and using whatever resourceVersion we already had. If we find we get into some nasty loop, we can always re-add it.

Reply:
We can't drop resetting the locally held RV here as well - put appropriate rationale in a comment in the code there as well.